cmd/releasego/update-azure-linux.go

```diff
@@ -0,0 +1,249 @@
+	pat := githubutil.BindPATFlag()
+
+	if err := p(); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	client, err := githubutil.NewClient(ctx, *pat)
+	if err != nil {
+		return err
+	}
+
+	assets, err := loadBuildAssets(buildAssetJSON)
+	if err != nil {
+		return err
+	}
+
+	golangSpecFileContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", golangSpecFilepath)
+	if err != nil {
+		return err
+	}
+
+	golangSpecFileContent, err = updateSpecFile(assets, golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	prevGoArchiveName, err := extractGoArchiveNameFromSpecFile(golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	// Validation (as described in previous response)
+	if assets.GoSrcURL == "" || assets.GoSrcHash == "" {
+		return fmt.Errorf("invalid or missing GoSrcURL or GoSrcHash in assets.json")
+	}
+
+	golangSignaturesFileContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", golangSignaturesFilepath)
+	if err != nil {
+		return err
+	}
+
+	golangSignaturesFileContent, err = updateSignatureFile(golangSignaturesFileContent, prevGoArchiveName, path.Base(assets.GoSrcURL), assets.GoSrcHash)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", cgManifestFilepath)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err = updateCGManifest(assets, cgManifestContent)
+	if err != nil {
+		return err
+	}
+
+	_ = golangSignaturesFileContent
+	_ = cgManifestContent
+
+	return nil
+}
+
+func loadBuildAssets(assetFilePath string) (*buildassets.BuildAssets, error) {
+	assets := new(buildassets.BuildAssets)
+
+	if err := stringutil.ReadJSONFile(assetFilePath, &assets); err != nil {
+		return nil, fmt.Errorf("error loading build assets: %w", err)
+	}
+
+	return assets, nil
+}
+
+func downloadFileFromRepo(ctx context.Context, client *github.Client, owner, repo, branch, filePath string) ([]byte, error) {
+	fileContent, err := githubutil.DownloadFile(ctx, client, owner, repo, branch, filePath)
+	if err != nil {
+		if errors.Is(err, githubutil.ErrNotExists) {
+			// Handle the specific case of the file not existing
+			return nil, fmt.Errorf("file '%s' not found in repository '%s' on branch '%s'", filePath, repo, branch)
+		} else {
+			// Handle other errors (network issues, authentication, etc.)
+			return nil, fmt.Errorf("failed to download file '%s': %w", filePath, err)
+		}
+	}
+
+	if len(fileContent) == 0 {
+		return nil, fmt.Errorf("downloaded file %s is empty", filePath)
+	}
+	return fileContent, nil
+}
+
+const (
+	golangSignaturesFilepath = "SPECS/golang/golang.signatures.json"
+	golangSpecFilepath       = "SPECS/golang/golang.spec"
+	cgManifestFilepath       = "cgmanifest.json"
+)
+
+var (
+	specFileGoFilenameRegex = regexp.MustCompile(`(%global ms_go_filename  )(.+)`)
+	specFileRevisionRegex   = regexp.MustCompile(`(%global ms_go_revision  )(.+)`)
```

Comment:
I don't think we should rely on number of spaces. The number of spaces is just because the original author of the spec file aligned some virtual columns, so the number could increase or decrease if another human touches it at any point and wants to realign. I think it's perfectly fine if this code even ignores the number of spaces and always puts in a single space in the PR update result.

(In general, I think we should keep an eye out for changes to the spec file that would make automated tooling easier to write, less fragile, or easier to review. My understanding is that we're free to make changes like that to the spec file. (I did some of that in https://github.com/microsoft/azurelinux/commit/3d42d8a38cc9dfe64c8b695c287d06c7eedfca4d.))